### PR TITLE
Turn backend improvements

### PR DESCRIPTION
**Pull request by cws1121**
##### SUMMARY
Contains:

1. A bug fix for WhatsApp template messages. Template strings for outgoing messages labeled as "invalid survey messages" were not getting parsed. These messages are now being converted to regular whatsapp template messages;
2. logic for calculating Infobip gateway fees, similar to whatsapp.

##### FEATURE FLAG
/

##### RISK ASSESSMENT / QA PLAN
Deployed and tested on staging. However, currently Infobip is not charging "Dimagi" domain for their services. Probably would be a good idea to test this one more time as soon as they begin to charge for the traffic and their services. 

##### PRODUCT DESCRIPTION
2. Infobip fees are being calculated based on delivery reports and message logs for each individual message (inbound and outbound) and each message is being converted to SmsBillable object.

### CODE REVIEW

**Comment by shubbless** [comment]
@proteusvacuum could you also take a look at this? We had some confusion around how to pass fees on messaging through- there was a feature flag /hq/flags/edit/enable_include_sms_gateway_charging/ that was only being used for one project @nemanja1040 can provide more context as well

**Review by millerdev — APPROVED** [review]
Tests would be nice.

File: corehq/messaging/smsbackends/infobip/models.py [hunk 1]
```diff
@@ -103,7 +107,20 @@ def _send_sms(self, config, to, msg, headers):
         }
         url = f'https://{config.personalized_subdomain}.{INFOBIP_DOMAIN}/sms/2/text/advanced'
         response = requests.post(url, json=payload, headers=headers)
-        return response.content
+        self.handle_response(response, msg)
+
+    def handle_response(self, response, msg):
+        if response.status_code != 200:
+            if response.status_code == 500:
```
Comment (proteusvacuum) [inline corehq/messaging/smsbackends/infobip/models.py:114]: nit: you could have put this as a separate condition outside of the `200` response to be a bit clearer.

**Comment by proteusvacuum** [comment]
Looks fine to me

**Pull request merged**